projects/plugins/boost/app/assets/src/js/stores/critical-css-recommendations.ts

```diff
@@ -97,52 +103,47 @@ export const dismissalError = writable( null );
  * @param {string} title Error display title.
  * @param {Object} error Error.
  */
-export function setDismissalError( title: string, error: JSONObject ): void {
-	dismissalError.set( {
+function setDismissalError( title: string, error: JSONObject ): void {
+	dismissalErrorStore.set( {
 		title,
 		error,
 	} );
 }
 
-/**
- * Dismiss the recommendation associated with the given provider key. Calls the
- * API to update the back end in lock-step.
- *
- * @param {string} key Key of recommendation to dismiss.
- */
-export async function dismissRecommendation( key: string ): Promise< void > {
-	await api.post( '/recommendations/dismiss', {
-		providerKey: key,
-		nonce: Jetpack_Boost.nonces[ 'recommendations/dismiss' ],
-	} );
-	dismissed.update( keys => [ ...keys, key ] );
-}
-
 /**
  * Clear all the dismissed recommendations.
  */
-export async function clearDismissedRecommendations(): Promise< void > {
+export async function clearDismissedIssues(): Promise< void > {
 	await api.post( '/recommendations/reset', {
 		nonce: Jetpack_Boost.nonces[ 'recommendations/reset' ],
 	} );
-	dismissed.set( [] );
+	issuesStore.update( issues => {
+		return issues.map( issue => {
+			issue.status = 'active';
+			return issue;
+		} );
+	} );
 }
 
 /**
  * Takes a Provider Key set of errors (in an object keyed by URL), and returns
  * a SortedErrorSet; an array which contains each type of error grouped. Also
  * groups things like HTTP errors by code.
  *
- * @param {Object} errors Errors in an object keyed by URL to group
+ * @param issue The recommendation the errors belong to.
  */
-function groupErrorsByFrequency( errors: {
-	[ url: string ]: CriticalCssErrorDetails;
-} ): ErrorSet[] {
-	const groupKeys = Object.values( errors ).map( groupKey );
+export function groupErrorsByFrequency( issue: CriticalCssIssue ): ErrorSet[] {
+	const { errors } = issue;
+	const groupKeys = errors.map( error => groupKey( error ) );
 	const groupOrder = sortByFrequency( groupKeys );
 
 	return groupOrder.map( group => {
-		const byUrl = objectFilter( errors, v => groupKey( v ) === group );
+		const byUrl = errors.reduce< { [ url: string ]: CriticalCssErrorDetails } >( ( acc, error ) => {
+			if ( groupKey( error ) === group ) {
+				acc[ error.url ] = error;
+			}
+			return acc;
+		}, {} );
 		const first = byUrl[ Object.keys( byUrl )[ 0 ] ];
 
```

Comment:
Not too clear what's going on in this function. Previously it wasn't clear either. But it would be nice to have some comments with examples.

Reply:
Yeah, I'll leave it as is at the moment, don't want to pull on too many threads in one go. This is not the end of refactor :)